### ISSUE TITLE
evaluation loss is nan

### ISSUE THREAD

**Issue by Mint-hfut** [meta-llama/llama-recipes#133]
### System Info

pytorch 2.0.1+cu117 cuda：11.7

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

When using the alpaca dataset, the data is truncated by the maximum length 224. If the instruction length is greater than 224, the labels will all be -100, which will cause loss to become nan during calculation. But strangely, there was no nan in the training phase with the loss.
Here is an example：
import torch
from torch.nn import CrossEntropyLoss
shift_logits = torch.randn([3,5])
shift_labels = torch.tensor([-100,-100,-100],dtype=torch.int64)
loss_fct = CrossEntropyLoss()
loss = loss_fct(shift_logits, shift_labels)
print(loss)

### Error logs

tensor(nan)

### Expected behavior

Whenever the number of tokens in instruction is greater than the maximum length, e.g. 224

**Comment by timothylimyl** [comment]
same issue, change the val batch and its okay, but val batch increase gpu mem even though same as training one.  Something seems off with the dataloader in regards to the batch sizing under multi-gpu and FSDP

**Comment by timothylimyl** [comment]
Another weird issue is that when I increase the val batch to be same as training, it goes OOM. Val batch somehow will take more than training.

**Comment by HamidShojanazeri** [comment]
This seems also to be solved, please feel free to re-open if still seeing the issue.
